### PR TITLE
repo: Lint the codebase

### PR DESCRIPTION
**Pull request by randomvariable**
Equalise with Cluster API Provider AWS in set of linters running, updating code to match.

Will allow all linters to be enabled as PR blocking, as well as developers being able
to run `make test` and `make lint`.

Not urgent for rc.1.

### CODE REVIEW

File: controllers/vspherevm_controller.go [hunk 1]
```diff
@@ -287,11 +288,9 @@ func (r vmReconciler) reconcileNormal(ctx *context.VMContext) (reconcile.Result,
 
 func (r vmReconciler) reconcileNetwork(ctx *context.VMContext, vm infrav1.VirtualMachine) {
 	ctx.VSphereVM.Status.Network = vm.Network
-	var ipAddrs []string
+	var ipAddrs []string // nolint:prealloc
```
Comment (detiber) [inline controllers/vspherevm_controller.go]:
```suggestion
	ipAddrs := make([]string, 0, len(vm.Network))
```

File: pkg/haproxy/haproxy_test.go [hunk 2]
```diff
@@ -101,20 +103,22 @@ func TestCreateLoadBalancer(t *testing.T) {
 	// Start a transaction.
 	var transactionID optional.String
 	g.Eventually(func() error {
-		txn, _, err := client.TransactionsApi.StartTransaction(ctx, version)
+		txn, resp, err := client.TransactionsApi.StartTransaction(ctx, version)
 		if err != nil {
 			return err
 		}
+		resp.Body.Close() // nolint:errcheck
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]:
```suggestion
		g.Expect(resp.Body.Close()).To(gomega.Succeed)
```

File: pkg/haproxy/haproxy_test.go [hunk 3]
```diff
@@ -91,7 +93,7 @@ func TestCreateLoadBalancer(t *testing.T) {
 	// Get the current configuration version.
 	var version int32
 	g.Eventually(func() error {
-		global, _, err := client.GlobalApi.GetGlobal(ctx, nil)
+		global, _, err := client.GlobalApi.GetGlobal(ctx, nil) // nolint:bodyclose
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]: Use the same close semantics as below?

File: pkg/haproxy/haproxy_test.go [hunk 4]
```diff
@@ -101,20 +103,22 @@ func TestCreateLoadBalancer(t *testing.T) {
...
 		}
+		resp.Body.Close() // nolint:errcheck
 		transactionID = optional.NewString(txn.Id)
 		return nil
 	}).ShouldNot(gomega.HaveOccurred(), "failed to start a transaction")
 
 	// Get a backend that does not exist.
-	_, _, err = client.BackendApi.GetBackend(ctx, "does-not-exist", nil)
+	_, resp, err := client.BackendApi.GetBackend(ctx, "does-not-exist", nil)
+	resp.Body.Close() // nolint:errcheck
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]:
```suggestion
		g.Expect(resp.Body.Close()).To(gomega.Succeed)
```

File: pkg/haproxy/haproxy_test.go [hunk 5]
```diff
@@ -124,20 +128,22 @@ func TestCreateLoadBalancer(t *testing.T) {
 	}, &hapi.CreateBackendOpts{
 		TransactionId: transactionID,
 	})
+	resp.Body.Close() // nolint:errcheck
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]:
```suggestion
		g.Expect(resp.Body.Close()).To(gomega.Succeed)
```

File: pkg/haproxy/haproxy_test.go [hunk 6]
```diff
@@ -124,20 +128,22 @@ func TestCreateLoadBalancer(t *testing.T) {
...
-	frontend, _, err := client.FrontendApi.CreateFrontend(ctx, hapi.Frontend{
+	frontend, resp, err := client.FrontendApi.CreateFrontend(ctx, hapi.Frontend{
 		Name:           "lb-frontend",
 		Mode:           haproxy.ModeTCP,
 		DefaultBackend: backend.Name,
 	}, &hapi.CreateFrontendOpts{
 		TransactionId: transactionID,
 	})
 	g.Expect(err).ToNot(gomega.HaveOccurred(), "failed to create frontend")
+	resp.Body.Close() // nolint:errcheck
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]:
```suggestion
		g.Expect(resp.Body.Close()).To(gomega.Succeed)
```

File: pkg/haproxy/haproxy_test.go [hunk 7]
```diff
@@ -148,13 +154,14 @@ func TestCreateLoadBalancer(t *testing.T) {
 			TransactionId: transactionID,
 		})
 	g.Expect(err).ToNot(gomega.HaveOccurred(), "failed to bind frontend")
+	resp.Body.Close() // nolint:errcheck
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]:
```suggestion
		g.Expect(resp.Body.Close()).To(gomega.Succeed)
```

File: pkg/haproxy/haproxy_test.go [hunk 8]
```diff
@@ -187,9 +194,10 @@ func TestCreateLoadBalancer(t *testing.T) {
 			TransactionId: transactionID,
 		})
 	g.Expect(err).ToNot(gomega.HaveOccurred(), "failed to bind create first backend server")
+	resp.Body.Close() // nolint:errcheck
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]:
```suggestion
		g.Expect(resp.Body.Close()).To(gomega.Succeed)
```

File: pkg/haproxy/haproxy_test.go [hunk 9]
```diff
@@ -202,27 +210,31 @@ func TestCreateLoadBalancer(t *testing.T) {
 			TransactionId: transactionID,
 		})
 	g.Expect(err).ToNot(gomega.HaveOccurred(), "failed to bind create second backend server")
+	resp.Body.Close() // nolint:errcheck
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]:
```suggestion
		g.Expect(resp.Body.Close()).To(gomega.Succeed)
```

File: pkg/haproxy/haproxy_test.go [hunk 10]
```diff
@@ -202,27 +210,31 @@ func TestCreateLoadBalancer(t *testing.T) {
...
 	// Commit the transaction.
-	_, _, err = client.TransactionsApi.CommitTransaction(
+	_, resp, err = client.TransactionsApi.CommitTransaction(
 		ctx,
 		transactionID.Value(),
 		&hapi.CommitTransactionOpts{
 			ForceReload: optional.NewBool(true),
 		})
 	g.Expect(err).ToNot(gomega.HaveOccurred(), "failed to commit the transaction")
+	resp.Body.Close() // nolint:errcheck
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]:
```suggestion
		g.Expect(resp.Body.Close()).To(gomega.Succeed)
```

File: pkg/haproxy/haproxy_test.go [hunk 11]
```diff
@@ -202,27 +210,31 @@ func TestCreateLoadBalancer(t *testing.T) {
...
 
 	// Get the current configurationv ersion.
-	global, _, err := client.GlobalApi.GetGlobal(ctx, nil)
+	global, resp, err := client.GlobalApi.GetGlobal(ctx, nil)
 	g.Expect(err).ToNot(gomega.HaveOccurred(), "failed get HAPI global config")
+	resp.Body.Close() // nolint:errcheck
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]:
```suggestion
		g.Expect(resp.Body.Close()).To(gomega.Succeed())
```

File: pkg/haproxy/haproxy_test.go [hunk 12]
```diff
@@ -202,27 +210,31 @@ func TestCreateLoadBalancer(t *testing.T) {
...
 
 	// Start a second transaction.
-	txn, _, err := client.TransactionsApi.StartTransaction(ctx, global.Version)
+	txn, resp, err := client.TransactionsApi.StartTransaction(ctx, global.Version)
 	g.Expect(err).ToNot(gomega.HaveOccurred(), "failed to start second transaction")
+	resp.Body.Close() // nolint:errcheck
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]:
```suggestion
		g.Expect(resp.Body.Close()).To(gomega.Succeed())
```

File: pkg/haproxy/haproxy_test.go [hunk 13]
```diff
@@ -233,11 +245,13 @@ func TestCreateLoadBalancer(t *testing.T) {
 		TransactionId: transactionID,
 	})
 	g.Expect(err).To(gomega.HaveOccurred())
+	resp.Body.Close() // nolint:errcheck
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]:
```suggestion
		g.Expect(resp.Body.Close()).To(gomega.Succeed())
```

File: pkg/haproxy/haproxy_test.go [hunk 14]
```diff
@@ -233,11 +245,13 @@ func TestCreateLoadBalancer(t *testing.T) {
...
 	g.Expect(haproxy.IsConflict(err)).To(gomega.BeTrue())
 
 	// Abandon the second transaction.
-	_, err = client.TransactionsApi.DeleteTransaction(ctx, transactionID.Value())
+	resp, err = client.TransactionsApi.DeleteTransaction(ctx, transactionID.Value())
 	g.Expect(err).ToNot(gomega.HaveOccurred())
+	resp.Body.Close() // nolint:errcheck
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]:
```suggestion
		g.Expect(resp.Body.Close()).To(gomega.Succeed())
```

File: pkg/haproxy/haproxy_test.go [hunk 15]
```diff
@@ -247,7 +261,7 @@ func TestCreateLoadBalancer(t *testing.T) {
 			Timeout:   time.Second * 1,
 			Transport: &http.Transport{},
 		}
-		_, err := client.Get(lbEndpoint)
+		_, err := client.Get(lbEndpoint) // nolint:bodyclose
```
Comment (detiber) [inline pkg/haproxy/haproxy_test.go]: Use same body close semantics as above?

File: test/e2e/capv_test_utils.go [hunk 16]
```diff
@@ -17,7 +17,7 @@ limitations under the License.
 package e2e
 
 import (
-	"crypto/sha1" //nolint:gosec
+	"crypto/sha1" // nolint:gosec
```
Comment (detiber) [inline test/e2e/capv_test_utils.go]: any reason not to just bump this to sha256?

File: test/e2e/govmomi_test.go [hunk 17]
```diff
@@ -89,23 +73,31 @@ func destroyVMsWithPrefix(prefix string) {
 	}
 }
 
-// nolint:deadcode
+// nolint:deadcode,unused
```
Comment (detiber) [inline test/e2e/govmomi_test.go]: Can we just remove this if it is unused? Or is it used and just not detected by the linter?

Comment (randomvariable) [reply test/e2e/govmomi_test.go]: my preference is too delete too, wasn't sure why it was marked like that in the first place. Gonna do it anyway I think

File: test/e2e/capv_test_utils.go (hunk 16 above)
Comment (randomvariable) [reply test/e2e/capv_test_utils.go]: this was just a bulk find and replace, didn't even spot this. changing it now.

**Pull request merged**